### ISSUE TITLE
NaN translates into -9223372036854775808m

### ISSUE THREAD

**Issue by yiyijin** [DirectXMan12/k8s-prometheus-adapter#280]
Hi all, 

I was able to see the custom metrics:
```
{
  "kind": "MetricValueList",
  "apiVersion": "custom.metrics.k8s.io/v1beta1",
  "metadata": {
    "selfLink": "/apis/custom.metrics.k8s.io/v1beta1/namespaces/test/pods/%2A/consumer-lag"
  },
  "items": [
    {
      "describedObject": {
        "kind": "Pod",
        "namespace": "test",
        "name": "pod-7bfdbdstnbp",
        "apiVersion": "/v1"
      },
      "metricName": "consumer-lag",
      "timestamp": "2020-03-18T21:44:17Z",
      "value": "30"
    },
    {
      "describedObject": {
        "kind": "Pod",
        "namespace": "test",
        "name": "pod-7bfdbdvvfqr",
        "apiVersion": "/v1"
      },
      "metricName": "consumer-lag",
      "timestamp": "2020-03-18T21:44:17Z",
      "value": "-9223372036854775808m"
    }
  ]
}
```
The **2nd** element in the items list, has value `-9223372036854775808m`

However, if I use the same query adapter used to query prometheus, I got this response, the **2nd** pod has value `NaN`, which is translated into `-9223372036854775808m`, any reason why?

```json
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "hostname": "pod-7bfdbdstnbp"
        },
        "value": [
          1584568664.816,
          "30"
        ]
      },
      {
        "metric": {
          "hostname": "pod-7bfdbdvvfqr"
        },
        "value": [
          1584568664.816,
          "NaN"
        ]
      }
    ]
  }
}
```

**Comment by Ladicle** [comment]
NaN multiplied by 1000 is still NaN. Also, the NaN of float64 is cast to int64 in the below code, so it overflows to 9223372036854775808. This value is handled as MilliQuantity, so the API returns -9223372036854775808m.
https://github.com/directxman12/k8s-prometheus-adapter/blob/4b5788e847e6b009d72b886f22d54f9363f9eba0/pkg/custom-provider/provider.go#L93

**Comment by shalomcbouncex** [comment]
Maybe having a flag/option to replace Nan values with 0 automatically, Kubernetes can't handle Nan obviously and this unexpected value cause it to scale to the maximum. In my case, when the histogram has not traffic the result ends up as Nan and I found myself hacking the query to end up with the same label set but with 0 instead of Nan.

**Comment by s-urbaniak** [comment]
fixed in #308
